src/cosalib/build.py

```diff
@@ -154,6 +160,28 @@ def clean(self):
             log.info(
                 'Removed temporary work directory at {}'.format(self.workdir))
 
+    def set_token(self):
+        """
+        Create a semaphore to claim the the platform exclusively.
+        """
+        tf = os.path.join(self.build_dir, f".{self.platform}.building")
```

Comment:
Not a blocker but would like to see these use `flock()` too, so that things properly clean up even if the outer process gets killed.  This could be as easy as invoking `flock` as a child process.

Reply:
Unless its a `SIGINT`, then we should be okay -- you're right a `flock()` would be nice too. I'll do that as a follow up PR.